app/services/BulletPointBuilder.scala

```diff
@@ -172,31 +192,42 @@ object BulletPointBuilder {
   def constructLeadingText(
       wordLimit: Int,
       texts: List[String],
+      textsProcessed: Int,
       matches: List[Match],
+      matchesProcessed: Int,
       items: List[String],
       matchText: Boolean,
       wordsProcessed: Int
   ): List[String] = {
 
     if (matchText) {
```

Comment:
like locateTextsAndMatchesContainingLeadingText() , this uses a boolean switch arg, to make the fn do two different things; work on texts or work on matches, would be clearer to create two fns

Reply:
I've split the function into two as requested.